docs/developer-docs/security/security-best-practices/dos.mdx

```diff
@@ -4,32 +4,72 @@ keywords: [security, concept, denial of service, dos, cycles, captcha]
 
-* Expensive calls that only need to be called from other canisters can require some amount of cycles to be sent along with the call to compensate for the cycles consumed by the execution.
+In the case of canisters, such attacks can quickly deplete cycles, making the canister unable to process legitimate requests. Learn more about DoS in the [Security Best Practices: Denial of Service](https://internetcomputer.org/docs/current/developer-docs/security/security-best-practices/dos) guide.
 
-* Finally, it is also an option to charge for ingress messages, but that is not currently supported by the protocol itself and a custom solution would need to be designed.
+### Preventing DoS and DDoS Attacks
 
-## For expensive calls, consider using captchas or proof of work
+To protect your canisters from DoS and DDoS attacks, consider the following strategies:
+- **Bot prevention techniques**: Use methods like captchas or proof of work to ensure only legitimate users can access your canister. Captchas help verify that the user is human, while proof of work requires the user to spend computational resources to proceed, deterring automated attacks. An example implementation can be found in the [Internet Identity project](https://github.com/dfinity/internet-identity).
+- **Monitor cycles usage**: Regularly track your canisters cycles consumption and set alerts for any sudden spikes that may indicate an attack.
+- **Ingress message charging**: While charging for ingress messages (external requests to the canister) is not natively supported, custom solutions could be implemented to make sure that any expensive actions have costs associated with them.
 
-### Security concern
+## 2. Managing Resource Allocation
 
-If an update or query call is expensive in terms of memory used or cycles consumed, this may make it easy for bots to render the canister unusable (e.g. by filling up storage).
+### The "Noisy Neighbor" Problem
 
-### Recommendation
+In a shared resource environment like the Internet Computer, multiple canisters can run on the same subnet. If one canister consumes too many resources (CPU, memory, etc.), it can negatively impact the performance of others on the same subnet. This is known as the "noisy neighbor" problem.
 
-If the dapp offers such operations, consider bot prevention techniques such as adding captchas or proof of work. An example captcha implementation can be found in [Internet Identity](https://github.com/dfinity/internet-identity).
+### Protecting Against Noisy Neighbors
+
+To mitigate the "noisy neighbor" issue, manage your canister's resource allocation effectively:
+- **Memory allocation**: You can set a memory reservation for your canister through the `memory_allocation` setting, ensuring that your canister has enough reserved memory to operate even if other canisters consume large amounts of resources.
```

Comment:
> enough reserved memory

it might be worthwhile pointing out that while the memory allocation guarantees a canister to have the specified amount of memory available, it also acts as an upper bound on the actual memory usage and messages will trap if they exceeded the specified memory allocation